src/Resolver/RedirectRouteResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Resolver;
+
+use function array_key_exists;
+use Exception;
+use const FILTER_SANITIZE_URL;
+use function filter_var;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class RedirectRouteResolver implements RedirectRouteResolverInterface
```

Comment:
The name of this is wrong. You are not resolving a route, you are resolving an URL